--- central.c
+++ central.c
@@ -28,6 +28,7 @@
 #include <zmk/events/battery_state_changed.h>
 #include <zmk/pointing/input_split.h>
 #include <zmk/hid_indicators_types.h>
+#include <zmk/events/split_central_status_changed.h>
 #include <zmk/physical_layouts.h>
 
 static int start_scanning(void);
@@ -1103,6 +1104,10 @@
 
     confirm_peripheral_slot_conn(conn);
     split_central_process_connection(conn);
+    raise_zmk_split_central_status_changed((struct zmk_split_central_status_changed){
+        .slot = peripheral_slot_index_for_conn(conn),
+        .connected = true,
+    });
     k_work_submit(&notify_status_work);
 }
 
@@ -1115,6 +1120,11 @@
 
     LOG_DBG("Disconnected: %s (reason %d)", addr, reason);
 
+    raise_zmk_split_central_status_changed((struct zmk_split_central_status_changed){
+        .slot = peripheral_slot_index_for_conn(conn),
+        .connected = false,
+    });
+
 #if IS_ENABLED(CONFIG_ZMK_SPLIT_BLE_CENTRAL_BATTERY_LEVEL_FETCHING)
     struct peripheral_event_wrapper ev = {
         .source = peripheral_slot_index_for_conn(conn),
